clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -608,6 +614,9 @@ public void onFailure(RuntimeException e) {
         @Override
         public void onComplete(ClientResponse response) {
             this.response = response;
+            // increment ref count to the client response in order to hold on to buffer till we are done with handler
+            // this will make sure everything except poll is released once fireCompleted call is done
+            this.response.incRefCount();
```

Comment:
As shown in the attached sequence diagram 
![MemoryPool](https://user-images.githubusercontent.com/1738460/127572588-7e9e0ff2-b38e-4784-8d63-858a1e58d143.png)

the ConsumerNetworkClient.poll method will execute both this incRefCount() and subsequently the decRefCount() inside fireCompletion(), what happens if we remove this pair of incRefCount and decRefCount?